kadmin/server.c

```diff
@@ -249,6 +249,30 @@ kadmind_dispatch(void *kadm_handlep, krb5_boolean initial,
 	krb5_store_int32(sp, ret);
 	break;
     }
+    case kadm_prune:{
+        op = "PRUNE";
+        ret = krb5_ret_principal(sp, &princ);
+        if(ret)
+            goto fail;
+        ret = krb5_ret_int32(sp, &prunekvno);
+        if(ret){
+            krb5_free_principal(contextp->context, princ);
+            goto fail;
+        }
+        krb5_unparse_name_fixed(contextp->context, princ, name, sizeof(name));
+        krb5_warnx(contextp->context, "%s: %s %s", client, op, name);
+        ret = _kadm5_acl_check_permission(contextp, KADM5_PRIV_CPW, princ);
+        if(ret){
+            krb5_free_principal(contextp->context, princ);
+            goto fail;
+        }
+        ret = kadm5_prune_principal(kadm_handlep, princ, prunekvno);
+        krb5_free_principal(contextp->context, princ);
+        krb5_storage_free(sp);
+        sp = krb5_storage_emem();
```

Comment:
Error checking is needed.

Reply:
code is copied from other features. i'm willing to work it out.

Reply:
Never mind, I'll fix this after.

Reply:
remarked as unresolved to keep track of the things i've promissed/asked to fix

Reply:
@nicowilliams I've added a simple check as requested (commit d82fffc). I did not check subsequent write to the storage object because a) it's not anywhere else, b) error handling code uses the same call, which probably would fail too in that case.

somewhat I feel that make it really right would require serious refactoring of the whole dispatch code :/ please review the added commit and resolve the conversation